diff --git a/modelunload.py b/modelunload.py
index ea86e8e..fb41a3b 100644
--- a/modelunload.py
+++ b/modelunload.py
@@ -7,11 +7,12 @@ class ModelUnloader:
     @classmethod
     def INPUT_TYPES(s):
          return {
-            "required": {			
+                 "required": { "image": ("IMAGE", )
             },
         }
 
-    RETURN_TYPES = ()
+    # RETURN_TYPES = ()
+    RETURN_TYPES = ("IMAGE",)
     RETURN_NAMES = ()
 
     FUNCTION = "unload_model"
@@ -20,7 +21,7 @@ class ModelUnloader:
 
     CATEGORY = "LJRE/utils"
 
-    def unload_model(self):
+    def unload_model(self, image):
         loadedmodels=model_management.current_loaded_models
         unloaded_model = False
         for i in range(len(loadedmodels) -1, -1, -1):
@@ -30,4 +31,4 @@ class ModelUnloader:
             unloaded_model = True
         if unloaded_model:
             model_management.soft_empty_cache()
-        return ()
+        return (image, )
